ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLintKLoggerInitializer.kt

```diff
@@ -1,58 +1,11 @@
 package com.pinterest.ktlint.core
 
-import ch.qos.logback.classic.Level
-import ch.qos.logback.classic.Logger
 import mu.KLogger
-import org.jetbrains.kotlin.utils.addToStdlib.ifTrue
-
-public enum class LogLevel { TRACE, DEBUG, INFO }
-
-public var logLevel: LogLevel = LogLevel.INFO
 
 @Deprecated("Environment variable is replace by new variables below")
 private const val KTLINT_DEBUG = "KTLINT_DEBUG"
 
-// Via command line parameters "--trace" and "--print-ast" the end user of ktlint can change the logging behavior. As
-// unit tests are not invoked via the main ktlint runtime, those command line parameters can not be used to change the
-// logging behavior while running the unit tests. Instead, the environment variables below can be used by ktlint
-// developers to change the logging behavior. Note, when set the environment variables also affect the runtinme of
-// ktlint. As of that the name of the variables start with KTLINT_UNIT_TEST to clarify the intent.
-public const val KTLINT_UNIT_TEST_DUMP_AST = "KTLINT_UNIT_TEST_DUMP_AST"
-public const val KTLINT_UNIT_TEST_TRACE = "KTLINT_UNIT_TEST_TRACE"
-public const val KTLINT_UNIT_TEST_ON_PROPERTY = "ON"
+public var loggerModifier: (KLogger) -> Unit = { _ -> }
```

Comment:
The solution where `loggerModifier` is passed as a parameter looks much cleaner, I agree.
However, the difference between these solutions is that if a modifier function is passed as a parameter - modification will be applied only to the current logger (one created by this call), but all others will still be created with the default logging level. If modifier represented by a variable - all loggers that will be created using `initKtLintKLogger` will also use the modifier method for configuration after creation.
In the general case, it is not a very good idea to make such a global configuration, but here it is justifiable in my opinion.

Reply:
Good catch. I forgot about this case.  I have now changed it so that:
* The default loggerModifier can only be set once.
* It is still possible to override the loggerModifier for a specific class by passing a loggerModifier to `initKtLintKLogger`.
* When `initKtLintKLogger` is called without parameter, the defaultLoggerModifier is applied when set.

```
/**
 * Default modifier for the KLogger of new instances of classes calling [initKtLintKLogger]. Classes for which
 * [initKtLintKLogger] has been called before setting this variable will not be changed. Also note, that this modifier
 * can only be set once.
 */
public lateinit var defaultLoggerModifier: (KLogger) -> Unit

/**
 * Initializes the logger with the [defaultLoggerModifier] when set.
 */
public fun KLogger.initKtLintKLogger(): KLogger {
    return if (::defaultLoggerModifier.isInitialized) {
        apply { defaultLoggerModifier(this) }
    } else {
        this
    }
}

/**
 * Initializes the logger with the [loggerModifier].
 */
public fun KLogger.initKtLintKLogger(
    loggerModifier: (KLogger) -> Unit
): KLogger = apply { loggerModifier(this) }
```